src/Config/ConfigurationStore.php

```diff
@@ -79,35 +79,57 @@ private function setFlags(array $flags): void
 
         try {
             $this->flagCache->setMultiple($serialized);
-        } catch (InvalidArgumentException $e) {
+        } catch (\Psr\SimpleCache\InvalidArgumentException $e) {
             // Simple cache throws exceptions when a keystring is not a legal value (characters {}()/@: are illegal)
             syslog(LOG_WARNING, "[EPPO SDK] Illegal flag key: " . $e->getMessage());
         }
     }
 
-    public function getFlagCacheAgeSeconds(): int
+    public function getMetadata(string $key): mixed
     {
         try {
-            $lastFetch = $this->metadataCache->get(self::FLAG_TIMESTAMP);
-            if ($lastFetch == null) {
-                return -1;
+            $meta = $this->metadataCache->get($key);
+            if ($meta != null) {
+                return unserialize($meta) ?: null; // unserialize returns false if there was a problem decoding.
             }
-        } catch (InvalidArgumentException $e) {
-            return -1;
+        } catch (\Psr\SimpleCache\InvalidArgumentException $e) {
+            syslog(LOG_WARNING, "[EPPO SDK] Illegal flag key: " . $e->getMessage());
         }
-        return time() - $lastFetch;
+
+        return null;
     }
 
     /**
      * @throws EppoClientException
      */
-    public function getBanditVariations(): BanditVariationIndexer
+    public function getBanditVariations(): IBanditVariationIndexer
     {
         try {
-            return unserialize($this->metadataCache->get(self::BANDIT_VARIATION_KEY));
-        } catch (InvalidArgumentException $e) {
+            $data = $this->metadataCache->get(self::BANDIT_VARIATION_KEY);
+            if ($data !== null) {
+                return unserialize($data);
+            }
+            return BanditVariationIndexer::empty();
+        } catch (\Psr\SimpleCache\InvalidArgumentException $e) {
             // We know that the key does not contain illegal characters so we should not end up here.
             throw EppoClientException::From($e);
         }
     }
+
+    /**
+     * @throws InvalidArgumentException
+     */
+    public function setMetadata(string $key, mixed $metadata): void
+    {
+        Validator::validateNotEqual(
+            $key,
+            self::BANDIT_VARIATION_KEY,
+            'Unable to use reserved key, ' . self::BANDIT_VARIATION_KEY
+        );
```

Comment:
Is this just being defensive? We control these keys right--a user won't be able to pick key for meta data?

Reply:
yep, you're right. All these layers got me thinking overly encapsulated.